src/vaults/legacy/LegacyVaultAdapter.sol

```diff
@@ -0,0 +1,205 @@
+    }
+
+    /// @inheritdoc IInvestmentManager
+    function maxMint(address, /* vault */ address user) public view returns (uint256 shares) {
+        shares = asyncManager().maxMint(address(this), user);
+    }
+
+    /// @inheritdoc IInvestmentManager
+    function maxWithdraw(address, /* vault */ address user) public view returns (uint256 assets) {
+        assets = asyncManager().maxWithdraw(address(this), user);
+    }
+
+    /// @inheritdoc IInvestmentManager
+    function maxRedeem(address, /* vault */ address user) public view returns (uint256 shares) {
+        shares = asyncManager().maxRedeem(address(this), user);
+    }
+
+    /// @inheritdoc IInvestmentManager
+    function pendingDepositRequest(address, /* vault */ address user) public view returns (uint256 assets) {
+        assets = asyncManager().pendingDepositRequest(address(this), user);
+    }
+
+    /// @inheritdoc IInvestmentManager
+    function pendingRedeemRequest(address, /* vault */ address user) public view returns (uint256 shares) {
+        shares = asyncManager().pendingRedeemRequest(address(this), user);
+    }
+
+    /// @inheritdoc IInvestmentManager
+    function pendingCancelDepositRequest(address, /* vault */ address user) public view returns (bool isPending) {
+        isPending = asyncManager().pendingCancelDepositRequest(address(this), user);
+    }
+
+    /// @inheritdoc IInvestmentManager
+    function pendingCancelRedeemRequest(address, /* vault */ address user) public view returns (bool isPending) {
+        isPending = asyncManager().pendingCancelRedeemRequest(address(this), user);
+    }
+
+    /// @inheritdoc IInvestmentManager
+    function claimableCancelDepositRequest(address, /* vault */ address user) public view returns (uint256 assets) {
+        assets = asyncManager().claimableCancelDepositRequest(address(this), user);
+    }
+
+    /// @inheritdoc IInvestmentManager
+    function claimableCancelRedeemRequest(address, /* vault */ address user) public view returns (uint256 shares) {
+        shares = asyncManager().claimableCancelRedeemRequest(address(this), user);
+    }
+
+    /// @inheritdoc IInvestmentManager
+    function priceLastUpdated(address /* vault */ ) public view returns (uint64 lastUpdated) {
+        lastUpdated = manager.priceLastUpdated(address(this));
+    }
+
+    // --- IInvestmentManager - Vault claim functions ---
+    /// @inheritdoc IInvestmentManager
+    function deposit(address, /* vault */ uint256 assets, address receiver, address owner)
+        public
+        legacy
+        returns (uint256 shares)
+    {
+        shares = asyncManager().deposit(address(this), assets, receiver, owner);
+    }
+
+    /// @inheritdoc IInvestmentManager
+    function mint(address, /* vault */ uint256 shares, address receiver, address owner)
+        public
+        legacy
+        returns (uint256 assets)
+    {
+        assets = asyncManager().mint(address(this), shares, receiver, owner);
```

Comment:
Should not be the vault?

```suggestion
        assets = asyncManager().mint(vault, shares, receiver, owner);
```

Reply:
Ahh, unless you want to forward to the async manager the adapter address to be able to later handle it

Reply:
```
    /// @inheritdoc IDepositManager
    function mint(address vaultAddr, uint256 shares, address receiver, address controller)
        public
        auth
        returns (uint256 assets)
    {
        AsyncInvestmentState storage state = investments[vaultAddr][controller];
        uint128 shares_ = shares.toUint128();
        _processDeposit(state, shares_, shares_, vaultAddr, receiver);

        assets = uint256(_calculateAssets(vaultAddr, shares_, state.depositPrice, MathLib.Rounding.Down));
    }
```

Luckily, there is never a token transfer between manager and vault. Always, between manager and escrow. Hence, we can shadow the legacy vault address completely from the new system through the adapter.